src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -511,23 +511,23 @@ private void writeRecord(SinkRecord record) throws IOException {
     long expectedOffset = offset + recordCounter;
     if (offset == -1) {
       offset = record.kafkaOffset();
-    } else if (record.kafkaOffset() != expectedOffset) {
+    } else if (record.kafkaOffset() < expectedOffset) {
```

Comment:
Well I think that  less-than condition will work for new version where this check is not needed at all, and is backwards compatible as well. It is essentially how the original condition should have been since it was expecting "rewinds" back in the partition but not "fast forwards" with holes.
Another option as you say is to just protect this under a configuration setting. That works for me as well. So you decide, if you want, I can change the PR to leave the original condition under a config flag.